src/tools/generate/generate_chapters.js

```diff
@@ -116,10 +119,39 @@ const generate_chapters = async (chapter_match) => {
 
 };
 
+const generate_syntex_snippet = (element, body, type) => {
+  const snippet_clean = element.innerHTML.replace(/&lt;/g, '<').replace(/&gt;/g, '>');
+  const snippet_converted = rainbow.colorSync(snippet_clean, type);
+  body = body.replace(element.innerHTML, snippet_converted);
+  return body;
+};
+
 const parse_file = async (markdown,chapter) => {
   const html = converter.makeHtml(markdown);
   let body = html;
 
+  // Syntax Highlighting 
+  const dom = new JSDOM(body);
+  const html_codes = dom.window.document.querySelectorAll('code.language-html');
+  const css_codes = dom.window.document.querySelectorAll('code.language-css');
+  const js_codes = dom.window.document.querySelectorAll('code.language-js');
+  const sql_codes = dom.window.document.querySelectorAll('code.language-sql');
+
+  html_codes.forEach(element => {
+    body = generate_syntex_snippet(element, body, "html");
+  });
+  css_codes.forEach(element => {
+    body = generate_syntex_snippet(element, body, "css");
+  });
+  js_codes.forEach(element => {
+    body = generate_syntex_snippet(element, body, "javascript");
+  });
+  sql_codes.forEach(element => {
+    body = generate_syntex_snippet(element, body, "sql");
+  });
```

Comment:
Could this all be moved to function so we could just call:

```
body = generate_syntax_snippet(element, body, "html");
body = generate_syntax_snippet(element, body, "css");
body = generate_syntax_snippet(element, body, "js");
body = generate_syntax_snippet(element, body, "sql");
```

And it does the `querySelectorAll` and `forEach` in the function? This would avoid having to repeat the same code 4 times.